dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -215,8 +216,11 @@ public Formula lastValue(MemoryObject base, int offset) {
         if (useIntegers) {
             return formulaManager.getIntegerFormulaManager().makeVariable(name);
         }
-        //TODO match this with the actual type.
-        return formulaManager.getBitvectorFormulaManager().makeVariable(base.getType().getBitWidth(), name);
+        //TODO match this with the actual type stored at the memory address
+        // (we do not know and guess the arch type right now)
+        TypeFactory types = TypeFactory.getInstance();
+        int archSize = types.getMemorySizeInBytes(types.getArchType());
+        return formulaManager.getBitvectorFormulaManager().makeVariable(archSize, name);
```

Comment:
This is only used for litmus tests. I this guessing a problem at all? Will it be once we have support for mixed size concurrency?

Reply:
If litmus tests declare variables of different sizes, it could be a problem. Also, it is unrelated to mixed-size concurrency. It's just that memory is currently untyped and as long as we support C-like mallocs memory will likely continue to be untyped. For example, if the memory content happened to be of float type, this code would fail.

Reply:
Litmus tests in the presence of mixed size concurrency will definitely define variables of different types, so this is definitely something we need to support once me move there

Reply:
Mixed-size has not much to do with this. These are the values at a specific memory address.
Since litmus code has only static memory allocations, we can know the exact type in memory.
A mixed-size access will not change the type of memory content, it will just overwrite the memory value partially.